apps/server/src/modules/task/repo/task.repo.ts

```diff
@@ -34,35 +39,21 @@ export class TaskRepo {
 		const [usersTasks, total] = await this.em.findAndCount(
 			Task,
 			{
+				// TODO task query builder, see NewsScope
 				$and: [
+					// TODO move into a logic group / director
 					{ id: { $nin: homeworksWithSubmissions } },
 					{ private: { $ne: true } },
 					{ course: { $in: coursesOfStudent } },
 					{ $or: [{ lesson: null }, { lesson: { $in: lessonsOfStudent } }] },
 					{ $or: [{ dueDate: { $gte: oneWeekAgo } }, { dueDate: null }] },
 				],
 			},
+			// TODO Populate in separate step
+			// TODO extract pagination, oderby
 			{ populate: ['course'], limit, offset: skip, orderBy: { dueDate: QueryOrder.ASC } }
 		);
 
-		const mappedTasks = usersTasks.map((task) => {
-			return {
-				id: task.id,
-				_id: task._id,
-				name: task.name,
-				duedate: task.dueDate,
-				courseName: task.course?.name,
-				displayColor: task.course?.color,
-				createdAt: task.createdAt,
-				updatedAt: task.updatedAt,
-			};
-		});
-
-		return {
-			data: mappedTasks,
-			total,
-			limit,
-			skip,
-		};
+		return [usersTasks, total];
```

Comment:
in the case of findAll() this would be an array. and it is now an array that contains the list of entities + plus their total number. i think this should be fine.